runner/config.go

```diff
@@ -151,6 +177,62 @@ func LoadConfig(p string, c *Config) error {
 		}
 	}
 
+	// Process metadata field - we support two notations for this field - either an object or
+	// an array of objects so we do the conversion here
+	if c.Metadata != nil {
+		ext := path.Ext(p)
+		if strings.EqualFold(ext, ".yaml") || strings.EqualFold(ext, ".yml") {
+			// Ensure that keys are of a string type and cast them
+			objData, isObjData2 := c.Metadata.(map[interface{}]interface{})
+			if isObjData2 {
+				nd := make(map[string]interface{})
+				for k, v := range objData {
+					sk, isString := k.(string)
+					if !isString {
+						return errors.New("Data key must string")
+					}
+					if len(sk) > 0 {
+						nd[sk] = v
+					}
+				}
+
+				c.Metadata = nd
+			} else {
+				// TODO: Refactor this into utility function
+				arrData, isArray := c.Metadata.([]interface{})
+
+				if isArray {
+					var array []map[string]interface{}
+					for _, item := range arrData {
+						objData3, isObjData3 := item.(map[interface{}]interface{})
+						newItem := make(map[string]interface{})
```

Comment:
I believe valid metadata object has to be `map[string]string` so here we should be asserting that both key and value in the internal representation of the objects in array are strings and then converting those to that type?